### PR TITLE
Make sure positive, then cast to uint to avoid warnings

### PR DESCRIPTION
**Pull request by emilbayes**

### CODE REVIEW

**Comment by emilbayes** [comment]
@mafintosh Can you just check that you agree this extra overhead is worth it?

File: src/macros.h [hunk 1]
```diff
@@ -86,7 +90,12 @@
   } \
   int64_t var = name->IntegerValue(); \
   \
-  if (var < min) { \
+  if (var < 0) { \
+    Nan::ThrowError(#var " must be at least 0"); \
+    return; \
+  } \
+  \
+  if ((uint64_t) var < min) { \
```
Comment (mafintosh) [inline src/macros.h]: Is it implied that min is always unsigned here?

Comment (mafintosh) [reply src/macros.h]: I prefer an extra paran around the cast and var also?

Comment (emilbayes) [reply src/macros.h]: Yes I think the macro name here is enough to imply that min is uint of some width. Will add parens around cast :)

**Pull request merged**